### PR TITLE
Switch to write-excel-file for table export

### PR DESCRIPTION
**Pull request by paulwarren-wk**
#### Reason for change

See #416 - a large part of the viewer JS file was the exceljs library. 

#### Description of change

Replace the use of `exceljs` with the much smaller `write-excel-file`.  This change reduces the size of the production `ixbrlviewer.js` from 1.8MiB to 834KiB.

#### Steps to Test

Open a report with tables in it, and confirm that the "Export Table" output works as it did before. 

**review**:
@Workiva/xt
@paulwarren-wk

### CODE REVIEW

File: iXBRLViewerPlugin/viewer/src/js/inspector.js [hunk 1]
```diff
@@ -155,7 +155,13 @@ export class Inspector {
 
     handleMessage(event) {
         const jsonString = event.originalEvent.data;
-        const data = JSON.parse(jsonString);
+        let data;
+        try {
+            data = JSON.parse(jsonString);
+        }
+        catch (e) {
+            return;
```
Comment (brettkail-wk) [inline iXBRLViewerPlugin/viewer/src/js/inspector.js:163]: Out of curiosity, why is this needed?  Does write-excel-file send messages internally to itself?

File: iXBRLViewerPlugin/viewer/src/js/tableExport.js [hunk 2]
```diff
@@ -243,10 +252,14 @@ export class TableExport {
...
-          FileSaver.saveAs( blob, 'table.xlsx');
+        const excelData = this._writeTable(data);
+        writeXlsxFile(excelData.data, {
+            columns: excelData.columns,
+            fileName: 'table.xlsx'
         });
+        //wb.xlsx.writeBuffer().then( data => {
+        //  const blob = new Blob( [data], {type: "application/octet-stream"} );
+        //  FileSaver.saveAs( blob, 'table.xlsx');
+        //});
```
Comment (brettkail-wk) [inline iXBRLViewerPlugin/viewer/src/js/tableExport.js:263]: Remove commented out?

**Review by brettkail-wk — APPROVED** [review]

File: iXBRLViewerPlugin/viewer/src/js/tableExport.js (hunk 2 above)
Comment (paulwarren-wk) [reply iXBRLViewerPlugin/viewer/src/js/tableExport.js]: Done in [36f1ffd](https://github.com/Workiva/ixbrl-viewer/commit/36f1ffd15a9ed59700864763cd7d3878067d0c2a)

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: iXBRLViewerPlugin/viewer/src/js/inspector.js (hunk 1 above)
Comment (austinmatherne-wk) [inline iXBRLViewerPlugin/viewer/src/js/inspector.js:163]: It would be good to add a comment here explaining that we're swallowing exceptions in order to ignore messages from the `write-excel-file` library

File: package.json [hunk 3]
```diff
@@ -44,7 +44,8 @@
     "less-loader": "^6.0.0",
     "lunr": "^2.3.8",
     "moment": "^2.24.0",
-    "postcss-less": "^6.0.0"
+    "postcss-less": "^6.0.0",
+    "write-excel-file": "^1.4.27"
```
Comment (austinmatherne-wk) [inline package.json:48]: We've begun committing the package lock file (it's now required for CI and our builds). Could you execute `npm install` and commit the revised package lock file?

Additionally, we should be able to eliminate `exceljs` and `file-saver` from our dependencies with these changes (the license file will also need to be updated).

File: iXBRLViewerPlugin/viewer/src/js/inspector.js (hunk 1 above)
Comment (paulwarren-wk) [reply iXBRLViewerPlugin/viewer/src/js/inspector.js:163]: Seemingly - something is creating 20 or so messages at the point that the file is downloaded.  I couldn't immediately make sense of it, but I think it might be coming from `jsZip` which `write-excel-file` depends on.

Comment (austinmatherne-wk) [reply iXBRLViewerPlugin/viewer/src/js/inspector.js:163]: It would be good to add a comment explaining that we're swallowing exceptions in order to ignore messages coming from the `write-excel-file` dependency.

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Review by austinmatherne-wk — DISMISSED** [review]

**Review by austinmatherne-wk — APPROVED** [review]
